### ISSUE TITLE
Make and MakeInstall does not work, the module evdi not loaded to the kernel

### ISSUE THREAD

**Issue by mehdi1005** [DisplayLink/evdi#503]
Yes
  * Are you using the latest driver?Yes
    
  * Are you using the latest EVDI version?Yes
    
  * If you are using a DisplayLink device, have you checked 'troubleshooting'
    on DisplayLink's website?Yes
    
  * Is this issue related to evdi/kernel?Yes
    

  * Linux distribution and its version I'm using FEDORA 41 with Cinamon
  * Linux kernel version  Kernel version 6.11.4-301.fc41.x86_64 but i also tried with 6.11.11.300.fc41.x86_64
  * Xorg version (if used)
  paul@fedora:~/Images/play/evdi-1.14.7$ sudo Xorg -version

X.Org X Server 1.21.1.14
X Protocol Version 11, Revision 0
Current Operating System: Linux fedora 6.11.4-301.fc41.x86_64 #1 SMP PREEMPT_DYNAMIC Sun Oct 20 15:02:33 UTC 2024 x86_64
Kernel command line: BOOT_IMAGE=(hd0,gpt2)/vmlinuz-6.11.4-301.fc41.x86_64 root=UUID=a378394d-6d76-49b2-9453-b775b58f50eb ro rootflags=subvol=root rd.luks.uuid=luks-a8348af4-3250-411f-acbc-0fd408cdee96 rhgb quiet
Build ID: xorg-x11-server 21.1.14-2.fc41 
Current version of pixman: 0.44.2
	Before reporting problems, check http://wiki.x.org
	to make sure that you have the latest version.

  * Desktop environment in use
paul@fedora:~/Images/play/evdi-1.14.7$ echo $XDG_CURRENT_DESKTOP
X-Cinnamon

Well hello Internet, how are you doing today? great me too.
I have an issue since 4 weeks, and i've been trying to solve it but it does not go anywhere so i need held.
The module EVDI is not loaded by the kernel and i have some errors :

```
paul@fedora:~/Images/play/evdi-1.14.7$ journalctl | grep evdi

y/evdi-1.14.7" cmd=6D6F6470726F6265202D762065766469 exe="/usr/bin/sudo" terminal=pts/0 res=success'
déc. 12 10:08:40 fedora sudo[5178]:     paul : TTY=pts/0 ; PWD=/home/paul/Images/play/evdi-1.14.7 ; USER=root ; COMMAND=/usr/sbin/modprobe -v evdi
déc. 12 10:08:41 fedora kernel: evdi: loading out-of-tree module taints kernel.
déc. 12 10:08:41 fedora kernel: evdi: module verification failed: signature and/or required key missing - tainting kernel
déc. 12 10:08:41 fedora kernel: evdi: [I] Initialising logging on level 4
déc. 12 10:08:41 fedora kernel: evdi: [I] Atomic driver: yes
déc. 12 10:08:41 fedora kernel: evdi: [I] Increasing device count to 4
déc. 12 10:08:41 fedora kernel: evdi: [I] (card0) Opened by Task 5196 (modprobe) of process 5196 (modprobe)
déc. 12 10:08:41 fedora kernel: evdi evdi.0: [drm] Cannot find any crtc or sizes
déc. 12 10:08:41 fedora kernel: [drm] Initialized evdi 1.14.7 for evdi.0 on minor 0
déc. 12 10:08:41 fedora kernel: evdi: [I] Evdi platform_device create
déc. 12 10:08:41 fedora kernel: evdi: [I] (card2) Opened by Task 5196 (modprobe) of process 5196 (modprobe)
déc. 12 10:08:41 fedora kernel: evdi evdi.1: [drm] Cannot find any crtc or sizes
déc. 12 10:08:41 fedora kernel: [drm] Initialized evdi 1.14.7 for evdi.1 on minor 2
déc. 12 10:08:41 fedora kernel: evdi: [I] Evdi platform_device create
déc. 12 10:08:41 fedora kernel: evdi: [I] (card3) Opened by Task 5196 (modprobe) of process 5196 (modprobe)
déc. 12 10:08:41 fedora kernel: evdi evdi.2: [drm] Cannot find any crtc or sizes
déc. 12 10:08:41 fedora kernel: [drm] Initialized evdi 1.14.7 for evdi.2 on minor 3
déc. 12 10:08:41 fedora kernel: evdi: [I] Evdi platform_device create
déc. 12 10:08:41 fedora kernel: evdi: [I] (card4) Opened by Task 5196 (modprobe) of process 5196 (modprobe)
déc. 12 10:08:41 fedora kernel: evdi evdi.3: [drm] Cannot find any crtc or sizes
déc. 12 10:08:41 fedora kernel: [drm] Initialized evdi 1.14.7 for evdi.3 on minor 4
déc. 12 10:08:41 fedora kernel: evdi: [I] Evdi platform_device create
déc. 12 10:08:41 fedora kernel: evdi: [I] (card0) Opened by Task 2134 (Xorg) of process 2134 (Xorg)
déc. 12 10:08:41 fedora kernel: evdi: [I] (card0) Closed by Task 2134 (Xorg) of process 2134 (Xorg)
déc. 12 10:08:41 fedora kernel: evdi: [I] (card0) Opened by Task 2134 (Xorg) of process 2134 (Xorg)
déc. 12 10:08:41 fedora kernel: evdi: [I] (card0) Closed by Task 2134 (Xorg) of process 2134 (Xorg)
déc. 12 10:08:41 fedora kernel: evdi: [I] (card0) Opened by Task 2134 (Xorg) of process 2134 (Xorg)
déc. 12 10:08:41 fedora kernel: evdi: [I] (card2) Opened by Task 2134 (Xorg) of process 2134 (Xorg)
déc. 12 10:08:41 fedora kernel: evdi: [I] (card2) Closed by Task 2134 (Xorg) of process 2134 (Xorg)
déc. 12 10:08:41 fedora kernel: evdi: [I] (card2) Opened by Task 2134 (Xorg) of process 2134 (Xorg)
déc. 12 10:08:41 fedora kernel: evdi: [I] (card2) Closed by Task 2134 (Xorg) of process 2134 (Xorg)
déc. 12 10:08:41 fedora kernel: evdi: [I] (card2) Opened by Task 2134 (Xorg) of process 2134 (Xorg)
déc. 12 10:08:41 fedora kernel: evdi: [I] (card3) Opened by Task 2134 (Xorg) of process 2134 (Xorg)
déc. 12 10:08:41 fedora kernel: evdi: [I] (card3) Closed by Task 2134 (Xorg) of process 2134 (Xorg)
déc. 12 10:08:41 fedora kernel: evdi: [I] (card3) Opened by Task 2134 (Xorg) of process 2134 (Xorg)
déc. 12 10:08:41 fedora kernel: evdi: [I] (card3) Closed by Task 2134 (Xorg) of process 2134 (Xorg)
déc. 12 10:08:41 fedora kernel: evdi: [I] (card3) Opened by Task 2134 (Xorg) of process 2134 (Xorg)
déc. 12 10:08:41 fedora kernel: evdi: [I] (card4) Opened by Task 2134 (Xorg) of process 2134 (Xorg)
déc. 12 10:08:41 fedora kernel: evdi: [I] (card4) Closed by Task 2134 (Xorg) of process 2134 (Xorg)
déc. 12 10:08:41 fedora kernel: evdi: [I] (card4) Opened by Task 2134 (Xorg) of process 2134 (Xorg)
déc. 12 10:08:41 fedora kernel: evdi: [I] (card4) Closed by Task 2134 (Xorg) of process 2134 (Xorg)
déc. 12 10:08:41 fedora kernel: evdi: [I] (card4) Opened by Task 2134 (Xorg) of process 2134 (Xorg)
```
`
When i go and see the logs file i have

```

`cat /var/lib/dkms/evdi/1.14.7/build/make.log

MODPOST /var/lib/dkms/evdi/1.14.7/build/Module.symvers
   scripts/mod/modpost -M        -o /var/lib/dkms/evdi/1.14.7/build/Module.symvers -n -T /var/lib/dkms/evdi/1.14.7/build/modules.order -i Module.symvers -e 
ERROR: modpost: module evdi uses symbol dma_buf_begin_cpu_access from namespace DMA_BUF, but does not import it.
ERROR: modpost: module evdi uses symbol dma_buf_vunmap from namespace DMA_BUF, but does not import it.
ERROR: modpost: module evdi uses symbol dma_buf_vmap from namespace DMA_BUF, but does not import it.
ERROR: modpost: module evdi uses symbol dma_buf_end_cpu_access from namespace DMA_BUF, but does not import it.
make[3]: *** [scripts/Makefile.modpost:145: /var/lib/dkms/evdi/1.14.7/build/Module.symvers] Error 1
make[2]: *** [/usr/src/kernels/6.11.4-301.fc41.x86_64/Makefile:1918: modpost] Error 2
make[1]: *** [Makefile:236: __sub-make] Error 2
make[1] : on quitte le répertoire « /usr/src/kernels/6.11.4-301.fc41.x86_64 »
make: *** [Makefile:38: all] Error 2`
```
I go and search for the files who contains de 'DMA_BUF' and they are all well imported in the files, also the option in the kernel is activated and i have no   missing dependencies or kernel headers.

**Comment by elguero** [comment]
Whoops. I just saw what you said for four weeks you have been having issues. Hmm.

The commit above was fixing something that was committed yesterday, which, will fix the evdi build issue for you.  Not sure what else may be going on?

Are you doing this by hand or using an RPM package? I see that you are on Fedora.

**Comment by mehdi1005** [comment]
Hi Michael,

Yes i see that there was an update yesterday, i use the lastest commit but i have the same issues.
Well, i use the RPM Package, but i had the same problems, so i do it by hand and i had the same again. I tried 3 different kernel on Fedora 41 but i have the same problem.

**Comment by elguero** [comment]
Yep, you are on rc2.  String literals are in rc3, unless they get removed before final release.

**Comment by 3F9VhYEEk** [comment]
This issue seems to be back in 6.13.4. I am having the same issue during the latest Fedora41 upgrade on my system.

From `cat /var/lib/dkms/evdi/1.14.7/build/make.log`:
```
In file included from /usr/src/kernels/6.13.4-200.fc41.x86_64/include/linux/module.h:22,
                 from evdi_drm_drv.h:16,
                 from evdi_gem.c:24:
evdi_gem.c:32:18: error: expected ‘,’ or ‘;’ before ‘DMA_BUF’
   32 | MODULE_IMPORT_NS(DMA_BUF);
      |                  ^~~~~~~
/usr/src/kernels/6.13.4-200.fc41.x86_64/include/linux/moduleparam.h:26:61: note: in definition of macro ‘__MODULE_INFO’
   26 |                 = __MODULE_INFO_PREFIX __stringify(tag) "=" info
      |                                                             ^~~~
/usr/src/kernels/6.13.4-200.fc41.x86_64/include/linux/module.h:299:33: note: in expansion of macro ‘MODULE_INFO’
  299 | #define MODULE_IMPORT_NS(ns)    MODULE_INFO(import_ns, ns)
      |                                 ^~~~~~~~~~~
evdi_gem.c:32:1: note: in expansion of macro ‘MODULE_IMPORT_NS’
   32 | MODULE_IMPORT_NS(DMA_BUF);
      | ^~~~~~~~~~~~~~~~
make[4]: *** [/usr/src/kernels/6.13.4-200.fc41.x86_64/scripts/Makefile.build:196: evdi_gem.o] Error 1
make[4]: *** Waiting for unfinished jobs....
In file included from /usr/src/kernels/6.13.4-200.fc41.x86_64/include/linux/module.h:22,
                 from /usr/src/kernels/6.13.4-200.fc41.x86_64/include/linux/device/driver.h:21,
                 from /usr/src/kernels/6.13.4-200.fc41.x86_64/include/linux/device.h:32,
                 from /usr/src/kernels/6.13.4-200.fc41.x86_64/include/linux/hdmi.h:28,
                 from /usr/src/kernels/6.13.4-200.fc41.x86_64/include/drm/drm_modes.h:30,
                 from /usr/src/kernels/6.13.4-200.fc41.x86_64/include/drm/drm_vblank.h:33,
                 from evdi_painter.c:15:
evdi_painter.c:40:18: error: expected ‘,’ or ‘;’ before ‘DMA_BUF’
   40 | MODULE_IMPORT_NS(DMA_BUF);
      |                  ^~~~~~~
/usr/src/kernels/6.13.4-200.fc41.x86_64/include/linux/moduleparam.h:26:61: note: in definition of macro ‘__MODULE_INFO’
   26 |                 = __MODULE_INFO_PREFIX __stringify(tag) "=" info
      |                                                             ^~~~
/usr/src/kernels/6.13.4-200.fc41.x86_64/include/linux/module.h:299:33: note: in expansion of macro ‘MODULE_INFO’
  299 | #define MODULE_IMPORT_NS(ns)    MODULE_INFO(import_ns, ns)
      |                                 ^~~~~~~~~~~
evdi_painter.c:40:1: note: in expansion of macro ‘MODULE_IMPORT_NS’
   40 | MODULE_IMPORT_NS(DMA_BUF);
      | ^~~~~~~~~~~~~~~~
```

**Comment by Harmos274** [comment]
Hello i may have the same problem as @3F9VhYEEk in 6.13.4-200.

```
In file included from /usr/src/kernels/6.13.4-200.fc41.x86_64/include/linux/module.h:22,
                 from evdi_drm_drv.h:16,
                 from evdi_gem.c:24:
evdi_gem.c:32:18: error: expected ‘,’ or ‘;’ before ‘DMA_BUF’
   32 | MODULE_IMPORT_NS(DMA_BUF);
      |                  ^~~~~~~
/usr/src/kernels/6.13.4-200.fc41.x86_64/include/linux/moduleparam.h:26:61: note: in definition of macro ‘__MODULE_INFO’
   26 |                 = __MODULE_INFO_PREFIX __stringify(tag) "=" info
      |                                                             ^~~~
/usr/src/kernels/6.13.4-200.fc41.x86_64/include/linux/module.h:299:33: note: in expansion of macro ‘MODULE_INFO’
  299 | #define MODULE_IMPORT_NS(ns)    MODULE_INFO(import_ns, ns)
      |                                 ^~~~~~~~~~~
evdi_gem.c:32:1: note: in expansion of macro ‘MODULE_IMPORT_NS’
   32 | MODULE_IMPORT_NS(DMA_BUF);
      | ^~~~~~~~~~~~~~~~
make[4]: *** [/usr/src/kernels/6.13.4-200.fc41.x86_64/scripts/Makefile.build:196: evdi_gem.o] Error 1
make[4]: *** Waiting for unfinished jobs....
In file included from /usr/src/kernels/6.13.4-200.fc41.x86_64/include/linux/module.h:22,
                 from /usr/src/kernels/6.13.4-200.fc41.x86_64/include/linux/device/driver.h:21,
                 from /usr/src/kernels/6.13.4-200.fc41.x86_64/include/linux/device.h:32,
                 from /usr/src/kernels/6.13.4-200.fc41.x86_64/include/linux/hdmi.h:28,
                 from /usr/src/kernels/6.13.4-200.fc41.x86_64/include/drm/drm_modes.h:30,
                 from /usr/src/kernels/6.13.4-200.fc41.x86_64/include/drm/drm_vblank.h:33,
                 from evdi_painter.c:15:
evdi_painter.c:40:18: error: expected ‘,’ or ‘;’ before ‘DMA_BUF’
   40 | MODULE_IMPORT_NS(DMA_BUF);
      |                  ^~~~~~~
/usr/src/kernels/6.13.4-200.fc41.x86_64/include/linux/moduleparam.h:26:61: note: in definition of macro ‘__MODULE_INFO’
   26 |                 = __MODULE_INFO_PREFIX __stringify(tag) "=" info
      |                                                             ^~~~
/usr/src/kernels/6.13.4-200.fc41.x86_64/include/linux/module.h:299:33: note: in expansion of macro ‘MODULE_INFO’
  299 | #define MODULE_IMPORT_NS(ns)    MODULE_INFO(import_ns, ns)
      |                                 ^~~~~~~~~~~
evdi_painter.c:40:1: note: in expansion of macro ‘MODULE_IMPORT_NS’
   40 | MODULE_IMPORT_NS(DMA_BUF);
      | ^~~~~~~~~~~~~~~~
make[4]: *** [/usr/src/kernels/6.13.4-200.fc41.x86_64/scripts/Makefile.build:196: evdi_painter.o] Error 1
make[3]: *** [/usr/src/kernels/6.13.4-200.fc41.x86_64/Makefile:2024: .] Error 2
make[2]: *** [/usr/src/kernels/6.13.4-200.fc41.x86_64/Makefile:263: __sub-make] Error 2
make[2]: Leaving directory '/var/lib/dkms/evdi/1.14.7/build'
make[1]: *** [Makefile:263: __sub-make] Error 2
make[1]: Leaving directory '/usr/src/kernels/6.13.4-200.fc41.x86_64'
make: *** [Makefile:38: all] Error 2

# exit code: 2
# elapsed time: 00:00:03
```

```
> uname -a 
Linux fedora 6.13.4-200.fc41.x86_64 #1 SMP PREEMPT_DYNAMIC Sat Feb 22 16:09:10 UTC 2025 x86_64 GNU/Linux
```

**Comment by synaptics-lspintzyk** [comment]
You are using evdi-1.14.7, this was fixed in evdi-1.14.8. You need to ask fedora package maintainer to update it.
